dataset_internal_test.go

```diff
@@ -0,0 +1,64 @@
+package ep
+
+import (
+	"github.com/panoplyio/ep/compare"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestAllMergeOptionResults(t *testing.T) {
+	t.Run("CompareResultEqualMergeOptions", func(t *testing.T) {
```

Comment:
if you do decide to leave it as separate test, name should be:
```suggestion
	t.Run("left is Equal", func(t *testing.T) {
```
which will produce testa named:
`TestMerge\left_is_Equal`
`TestMerge\left_is_BothNulls`
`TestMerge\left_is_Null`
...